Pidaq/UI/DTS/dts_dash/callbacks/control_callbacks.py

```diff
@@ -0,0 +1,198 @@
+            logger.warning("NO NAME SELECTED - PLEASE SELECT A PROFILE NAME BEFORE ADDING COMMANDS") 
+            ret_value = no_update, no_update, no_update, True
+            return ret_value
+            
+        # Warn if new command makes test length > 60 secs (60 000ms)
+        current_test_time = sum([command.step for command in test_profile.commands])
+        new_total_test_time = step + current_test_time
+        
+        if new_total_test_time > 60000:
+            logger.warn(f"TOTAL TEST TIME OF {new_total_test_time}ms / {new_total_test_time/1000}s EXCEEDS 1 MINUTE")
+            toggle_time_warning = True
+        else:
+            toggle_time_warning = False
+            
+        testType = test_profile.testType
+        command = DtsCommand(testType, step, value)
+        test_profile.addCommand(command)
+    
+        logger.info(f"COMMAND ADDED: {test_profile.commands[-1]}")
+        ret_value = test_profile.getDf().to_dict('records'), no_update, toggle_time_warning, no_update
+        
+    elif button_id == "clear-all-btn":
+        test_profile.clearAll()
+        ret_value = test_profile.getDf().to_dict('records'), no_update, no_update, no_update
+        logger.info(f"CLEAR ALL PRESSED - CLEARED ALL COMMANDS")
+    
+    elif button_id == "clear-last-btn":
+        removed_cmd = test_profile.clearLast()
+        ret_value = test_profile.getDf().to_dict('records'), no_update, no_update, no_update
+        logger.info(f"CLEAR LAST PRESSED - REMOVED COMMAND: {removed_cmd}")
+   
+    elif button_id == "rpm-toggle-btn":
+        
+        # If test type wasnt rpm, update test profile to rpm and clear commands 
+        if test_profile.testType != DtsTestType.RPM:
+            test_profile.testType = DtsTestType.RPM
+            test_profile.commands = []
+            logger.info(f"Test type updated to RPM, updated " 
+                        f"profile test type to: {test_profile.testType}")
+            
+            ret_value = test_profile.getDf().to_dict('records'), "RPM", no_update, no_update
+            
+        # If test profile type is already rpm - do nothing
+        else:
+            logging.info("RPM TOGGLE CLICKED - TEST TYPE IS ALREADY RPM - NO ACTION TAKEN ")
+            raise dash.exceptions.PreventUpdate
+        
+    elif button_id == "torque-toggle-btn":
+        
+        if test_profile.testType != DtsTestType.TORQUE:
+            test_profile.testType = DtsTestType.TORQUE
+            test_profile.commands = []
+            logger.info(f"Test type updated to torque, updated " 
+                        f"profile test type to: {test_profile.testType}")
+            
+            ret_value = test_profile.getDf().to_dict('records'), "TORQUE", no_update, no_update
+        
+        else:
+            logging.info("TORQUE TOGGLE CLICKED - TEST TYPE IS ALREADY TORQUE - NO ACTION TAKEN ")
+            raise dash.exceptions.PreventUpdate
+            
+    return ret_value
```

Comment:
A lot of lines here exceed 80 chars - I was thinking fixing it may sacrifice some readability but I'm considering it

Reply:
Up to you, not a huge deal though